com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/ValidateDeployParametersStep.java

```diff
@@ -48,32 +58,70 @@ protected String getStepErrorMessage(DelegateExecution context) {
     }
 
     private void validateParameters(DelegateExecution context) {
-        validateStartTimeout(context);
-        validateAppArchiveId(context);
-        validateExtDescriptorFileId(context);
-        validateVersionRule(context);
+        try {
+            validateStartTimeout(context);
+            validateAppArchiveId(context);
+            validateExtDescriptorFileId(context);
+            validateVersionRule(context);
+            verifyArchiveSignature(context);
+        } finally {
+            deleteArchiveFile(context);
+        }
     }
 
-    private void validateVersionRule(DelegateExecution context) {
-        String versionRuleString = (String) context.getVariable(Constants.PARAM_VERSION_RULE);
+    private void validateStartTimeout(DelegateExecution context) {
+        Object parameter = context.getVariable(Constants.PARAM_START_TIMEOUT);
+        if (parameter == null) {
+            return;
+        }
+        int startTimeout = (int) parameter;
+        if (startTimeout < 0) {
+            throw new SLException(Messages.ERROR_PARAMETER_1_MUST_NOT_BE_NEGATIVE, startTimeout, Constants.PARAM_START_TIMEOUT);
+        }
+    }
+
+    private void validateAppArchiveId(DelegateExecution context) {
+        String appArchiveId = StepsUtil.getRequiredString(context, Constants.PARAM_APP_ARCHIVE_ID);
+        String[] appArchivePartsId = appArchiveId.split(",");
+        List<FileEntry> archivePartEntries = getArchivePartEntries(context, appArchivePartsId);
+        StepsUtil.setAsJsonBinaries(context, Constants.VAR_FILE_ENTRIES, archivePartEntries);
+        getStepLogger().debug(Messages.BUILDING_ARCHIVE_FROM_PARTS);
+        new ExecutionRetrier().executeWithRetry(createArchiveFromParts(context, archivePartEntries));
+    }
+
+    private List<FileEntry> getArchivePartEntries(DelegateExecution context, String[] appArchivePartsId) {
+        return Arrays.stream(appArchivePartsId)
+                     .map(appArchivePartId -> findFile(context, appArchivePartId))
+                     .collect(Collectors.toList());
+    }
+
+    private Runnable createArchiveFromParts(DelegateExecution context, List<FileEntry> archivePartEntries) {
+        return () -> {
+            String archiveFileName = new ArchiveMerger(fileService, getStepLogger(), context).createArchiveFromParts(archivePartEntries);
+            context.setVariable(Constants.VAR_ARCHIVE_FILE_NAME, archiveFileName);
```

Comment:
Setting a Flowable variable in order to read it in the same step and not use it anywhere else is a code smell. Think of some other way to accomplish this.